modules/dqt/jsx/react.filterBuilder.js

```diff
@@ -592,6 +592,7 @@ class FilterGroup extends Component {
             <FilterGroup group={child}
                          items={this.props.items}
                          index={index}
+                         updateFilter={this.updateChild}
```

Comment:
This line fixes the bug. The other code change is just cleanup I saw when debugging.

Reply:
What is "the bug"?

Reply:
@driusan The bug occurs when deleting the first rule of the filter, then making a Group rule and attempting to use the buttons (Add Rule, Add Group, Delete Group) of the newly created Group Rule.
Console output of error:
`TypeError: this.props.updateFilter is not a function. (In 'this.props.updateFilter(group)', 'this.props.updateFilter' is undefined)`

Reply:
@driusan also should note that @pierre-p-s found that current fix solves the buttons from not working but the query results strangely aren't correct. I believe the query returns the correct results for the built filter when the query is saved and loaded. So I'm adding needs work tag until I can debug that part.